File: app/src/main/java/com/gee12/mytetroid/adapters/RecordsListAdapter.java
package com.gee12.mytetroid.adapters;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import com.gee12.mytetroid.R;
import com.gee12.mytetroid.data.SettingsManager;
import com.gee12.mytetroid.fragments.MainPageFragment;
import com.gee12.mytetroid.model.TetroidRecord;

import java.util.ArrayList;
import java.util.List;

public class RecordsListAdapter extends RecordsBaseListAdapter {

    private List<TetroidRecord> dataSet;

    public RecordsListAdapter(Context context, OnRecordAttachmentClickListener onAttachmentClickListener) {
        super(context, onAttachmentClickListener);
        this.dataSet = new ArrayList<>();
    }

    public void setDataItems(List<TetroidRecord> dataSet, int viewId) {
        this.dataSet = dataSet;
        this.isShowNodeName = (viewId == MainPageFragment.MAIN_VIEW_TAG_RECORDS
                || viewId == MainPageFragment.MAIN_VIEW_FAVORITES);
        this.dateTimeFormat = SettingsManager.checkDateFormatString(context);
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return dataSet.size();
    }

    @Override
    public Object getItem(int position) {
        return dataSet.get(position);
    }

    @Override
    public long getItemId(int position) {
        return dataSet.get(position).getId().hashCode();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        RecordViewHolder viewHolder;
        if (convertView == null) {
            viewHolder = new RecordViewHolder();
            convertView = inflater.inflate(R.layout.list_item_record, null);
            viewHolder.lineNumView = convertView.findViewById(R.id.record_view_line_num);
            viewHolder.iconView = convertView.findViewById(R.id.record_view_icon);
            viewHolder.nameView = convertView.findViewById(R.id.record_view_name);
            viewHolder.nodeNameView = convertView.findViewById(R.id.record_view_node);
            viewHolder.authorView = convertView.findViewById(R.id.record_view_author);
            viewHolder.tagsView = convertView.findViewById(R.id.record_view_tags);
            viewHolder.createdView = convertView.findViewById(R.id.record_view_created);
            viewHolder.editedView = convertView.findViewById(R.id.record_view_edited);
            viewHolder.attachedView = convertView.findViewById(R.id.record_view_attached);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (RecordViewHolder) convertView.getTag();
        }
        prepareView(position, viewHolder, convertView, dataSet.get(position));
        return convertView;
    }

    public List<TetroidRecord> getDataSet() {
        return dataSet;
    }
}

File: app/src/main/java/com/gee12/mytetroid/ui/main/RecordsListAdapter.kt
package com.gee12.mytetroid.ui.main

import android.content.Context
import android.view.View
import android.view.ViewGroup
import com.gee12.mytetroid.R
import com.gee12.mytetroid.common.Constants
import com.gee12.mytetroid.providers.IResourcesProvider
import com.gee12.mytetroid.model.TetroidRecord
import java.util.*
import kotlin.collections.ArrayList

class RecordsListAdapter(
    context: Context,
    resourcesProvider: IResourcesProvider,
    dateTimeFormat: String,
    getEditedDateCallback: (record: TetroidRecord) -> Date?,
    onClick: (record: TetroidRecord) -> Unit,
) : RecordsBaseListAdapter(
    context,
    resourcesProvider,
    dateTimeFormat,
    getEditedDateCallback,
    onClick
) {
    var dataSet: List<TetroidRecord>
        private set

    init {
        dataSet = ArrayList()
    }

    fun reset() {
        dataSet = ArrayList()
        notifyDataSetChanged()
    }

    fun setDataItems(dataSet: List<TetroidRecord>, viewId: Int) {
        this.dataSet = dataSet
        isShowNodeName = viewId in arrayOf(Constants.MAIN_VIEW_TAG_RECORDS, Constants.MAIN_VIEW_FAVORITES)
        notifyDataSetChanged()
    }

    fun delete(record: TetroidRecord) {
        (dataSet as MutableSet<*>).remove(record)
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return dataSet.size
    }

    override fun getItem(position: Int): Any {
        return dataSet[position]
    }

    override fun getItemId(position: Int): Long {
        return dataSet[position].id.hashCode().toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        val viewHolder: RecordViewHolder
        if (view == null) {
            viewHolder = RecordViewHolder()
            view = inflater.inflate(R.layout.list_item_record, null)
            viewHolder.lineNumView = view.findViewById(R.id.record_view_line_num)
            viewHolder.iconView = view.findViewById(R.id.record_view_icon)
            viewHolder.nameView = view.findViewById(R.id.record_view_name)
            viewHolder.nodeNameView = view.findViewById(R.id.record_view_node)
            viewHolder.authorView = view.findViewById(R.id.record_view_author)
            viewHolder.tagsView = view.findViewById(R.id.record_view_tags)
            viewHolder.createdView = view.findViewById(R.id.record_view_created)
            viewHolder.editedView = view.findViewById(R.id.record_view_edited)
            viewHolder.attachedView = view.findViewById(R.id.record_view_attached)
            view.tag = viewHolder
        } else {
            viewHolder = view.tag as RecordViewHolder
        }
        prepareView(position, viewHolder, view!!, dataSet[position])

        return view
    }
}